### DISCUSSION TITLE
v6 HasOneCreateAssociationMixin ignore the foreign key

### DISCUSSION THREAD

**Discussion by ghost** [sequelize/sequelize#16690] [Help & Questions]
In v6, the `HasManyCreateAssociationMixin` lets you ignore the foreign key, but the `HasOneCreateAssociationMixin` does not. Is there any way around this?

e.g. I have this in my Model class:

```ts
  declare createUserRole: Sequelize.HasManyCreateAssociationMixin<
    UserRole,
    "user_id"
  >;
```

which lets me call it without specifying the `user_id` since the parent record `user` already has it:

```ts
        await user.createUserRole( role_id: Role.Ids.guest });
```

But I can't seem to find a way to do that here:

```ts
  declare createUserAddress: Sequelize.HasOneCreateAssociationMixin<UserAddress>;
```

so that I can do:

```ts
        await user.createUserAddress( ...valuesWithoutUserId });
```

**Comment by ephys** [comment]
This was fixed in https://github.com/sequelize/sequelize/pull/15756, in v7
